raiden/utils/nursery.py

```diff
@@ -139,20 +147,29 @@ def subprocess_stopped(result: AsyncResult) -> None:
             def spawn_under_watch(function: Callable, *args: Any, **kwargs: Any) -> Greenlet:
                 greenlet = gevent.spawn(function, *args, **kwargs)
 
-                def kill_grenlet(_result: AsyncResult) -> None:
-                    # gevent.GreenletExit must **not** be used here, since that
-                    # exception is considered a successful run and it is not
-                    # re-raised on calls to `get`
-                    exception = RuntimeError("Janitor is stopping")
-                    greenlet.throw(exception)
-
-                # The Event.rawlink is executed inside the Hub thread, which
-                # does validation and *raises on blocking calls*, to go around
-                # this a new greenlet has to be spawned, that in turn will
-                # raise the exception.
-                callback = SpawnedLink(kill_grenlet)
```

Comment:
This was actually wrong, I re-read the documentation of `throw`:


        Should only be called from the HUB, otherwise the current greenlet is left unscheduled forever.
        To raise an exception in a safe manner from any greenlet, use :meth:`kill`.

https://www.gevent.org/_modules/gevent/greenlet.html

An alternative would be to use `kill` instead, so far I have not seen the problem the docs say can happen, but I did the change just to prevent bugs.